### PR TITLE
DataContainer class and version check

### PR DESCRIPTION
**Pull request by kbwestfall**
Addresses #934 .  Changes are effectively all in `datamodel.py`

Tests pass (see below), but it would be useful if someone took a look at the Gemini GNIRS pisco 1D coadd.  I had to replace a number of files in the test directory (including a shane_kast_red one, as well) for the tests to pass.  But the issue when running the coadd1d test is that I got a lot of `RuntimeWarning` messages, likely indicating that there are some NaN or infs in the output.  If you've seen that before, feel free to ignore this.  I just wanted to bring it up.

I haven't run the DevSuite on this branch, but can if requested.

```
test_alignments.py .                                                     [  0%]
test_arc.py .                                                            [  0%]
test_arcimage.py ..                                                      [  1%]
test_biasframe.py ..                                                     [  2%]
test_bitmask.py ....                                                     [  4%]
test_bpmimage.py ..                                                      [  5%]
test_bspline.py ..........                                               [  9%]
test_buildimage.py .                                                     [ 10%]
test_calibrations.py .......                                             [ 13%]
test_coadd.py .                                                          [ 13%]
test_cooked.py .                                                         [ 13%]
test_coremeta.py .                                                       [ 14%]
test_datacontainer.py ........                                           [ 17%]
test_detector.py ...                                                     [ 19%]
test_edgetrace.py .                                                      [ 19%]
test_flatfield.py .                                                      [ 20%]
test_flexure.py .                                                        [ 20%]
test_flux.py ....                                                        [ 22%]
test_fluxspec.py ..                                                      [ 23%]
test_load.py .                                                           [ 23%]
test_load_images.py ...........                                          [ 28%]
test_masterframe.py .                                                    [ 28%]
test_match.py ....                                                       [ 30%]
test_metadata.py .....                                                   [ 32%]
test_moment.py ......                                                    [ 35%]
test_msgs.py ..                                                          [ 36%]
test_opticalmodel.py ....                                                [ 37%]
test_parse.py ...                                                        [ 39%]
test_pca.py ....                                                         [ 41%]
test_processimage.py ..                                                  [ 41%]
test_procimg.py .                                                        [ 42%]
test_pydl.py .                                                           [ 42%]
test_pypeit.py ..                                                        [ 43%]
test_pypeitimage.py .                                                    [ 44%]
test_pypeitpar.py .................................                      [ 58%]
test_pypeitsetup.py .......                                              [ 61%]
test_qa.py .                                                             [ 62%]
test_runpypeit.py ..                                                     [ 62%]
test_scienceimage.py ...                                                 [ 64%]
test_scripts.py .........                                                [ 68%]
test_setups.py ................                                          [ 75%]
test_skysub.py ..                                                        [ 75%]
test_slitmask.py .                                                       [ 76%]
test_slittrace.py ....                                                   [ 78%]
test_spec2dobj.py ......                                                 [ 80%]
test_specobj.py ......                                                   [ 83%]
test_specobjs.py .....                                                   [ 85%]
test_spectrographs.py ....................                               [ 94%]
test_syncspec.py .                                                       [ 94%]
test_traceimage.py .                                                     [ 95%]
test_utils.py ....                                                       [ 96%]
test_vcorr.py ..                                                         [ 97%]
test_wavemodel.py .                                                      [ 98%]
test_wavetilts.py ....                                                   [100%]

======================= 229 passed in 521.88s (0:08:41) ========================
```

### CODE REVIEW

**Comment by kbwestfall** [comment]
Okay, I think I have it now.  Tests still pass, and:

```
$ python3
Python 3.7.6 (v3.7.6:43364a7ae0, Dec 18 2019, 14:18:50)
[Clang 6.0 (clang-600.0.57)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from pypeit import specobjs
>>> sobjs = specobjs.SpecObjs.from_fitsfile('pisco_coadd.fits')

[ERROR]   :: The HDU(s) cannot be parsed by a SpecObj object!
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/westfall/Work/packages/pypeit/pypeit/specobjs.py", line 80, in from_fitsfile
    sobj = specobj.SpecObj.from_hdu(hdu)
  File "/Users/westfall/Work/packages/pypeit/pypeit/datamodel.py", line 1188, in from_hdu
    msgs.error('The HDU(s) cannot be parsed by a {0} object!'.format(cls.__name__))
  File "/Users/westfall/Work/packages/pypeit/pypeit/pypmsgs.py", line 267, in error
    raise PypeItError(msg)
pypeit.pypmsgs.PypeItError: The HDU(s) cannot be parsed by a SpecObj object!
>>>
```

**Comment by profxj** [comment]
https://travis-ci.org/github/pypeit/PypeIt/builds/701812320

File: pypeit/datamodel.py [hunk 1]
```diff
@@ -902,11 +902,15 @@ def _parse(cls, hdu, ext=None, transpose_table_arrays=False, debug=False,
         else:
             _hdu = hdu
 
-        _ext = np.atleast_1d(ext)
+        _ext = np.atleast_1d(np.array(ext, dtype=object))
+        str_ext = np.logical_not([isinstance(e, (int, np.integer)) for e in _ext])
```
Comment (jhennawi) [inline pypeit/datamodel.py:906]: So are we switching from np.invert to np.logical_not. I actually don't know which one is preferred but thought they were equivalent. If so, maybe allowing both is fine. Would be good to know the answer though.

Comment (kbwestfall) [reply pypeit/datamodel.py:906]: @rcooke-ast is right that we should be switching.  `logical_not` is robust to data type:
```
$ python3
Python 3.7.6 (v3.7.6:43364a7ae0, Dec 18 2019, 14:18:50)
[Clang 6.0 (clang-600.0.57)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import numpy
>>> t = numpy.array([1, 0])
>>> numpy.logical_not(t)
array([False,  True])
>>> numpy.invert(t)
array([-2, -1])
>>> _t = numpy.array([True, False])
>>> numpy.logical_not(_t)
array([False,  True])
>>> numpy.invert(_t)
array([False,  True])
```

In practice, though, I don't know when or if we have used `invert` on an integer array.

**Comment by profxj** [comment]
merging.

**Pull request merged**